.expeditor/buildkite/test_core_plan_linux.rb

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+require 'Open3'
+require 'optparse'
+
+# Helper script to build and test any Habitat Core Plan
+
+def show_run_cmd(cmd, header=nil)
+  p "=====> #{header}" if header
+  p "-> #{cmd}"
+  # can explore other options here as needed e.g. adding a logger or improving the shell callout
+  res = system(cmd)
+  abort("Command '#{cmd}' failed with result '#{res}'") unless res
+end
+
+options = {}
+OptionParser.new do |opts|
+  opts.banner = 'Usage: build_test_habitat_core_plan.rb [options]'
+  opts.on('-p', '--package=PACKAGE', 'Local Habitat Core Package directory path to run against.') do |p|
+    options[:package] = p
+  end
+end.parse!
+
+package = options[:package]
+abort("Package directory must be specified!") unless package
+abort("Package directory #{package} is not a directory!") unless File.directory?(package)
+abort("Package directory does not have InSpec config") unless File.file?(File.join(package, 'inspec.yml'))
+
+prefix = "cd #{File.expand_path(package)} && "
+show_run_cmd("#{prefix}hab pkg build .", header="Building package: #{package}")
+show_run_cmd("#{prefix}. results/last_build.env && hab pkg export docker \"results/${pkg_artifact}\"", header="Exporting Docker image for package: #{package}")
```

Comment:
The pkg build on line 30 will happen inside a studio.
We should ideally just execute the tests on the CI host and not need to use a container. But due care is needed to make sure that the CI host  is in a clean state.